### DISCUSSION TITLE
About storage for GroupState

### DISCUSSION THREAD

**Discussion by streetycat** [buckyos/CYFS#97] [Ideas]
In the development process of "Group", I need an storage module based on the CYFS kernel. I have several requirements:

1. It can store the kernel information of the "Group" module, including APP status information and consensus chain information. The specific content is described in:
[https://github.com/buckyos/CYFS/issues/71#issuecomment-1378616109](url)

2. Similar to RootState, the objects mounted on it will not be recycled

3. Authority:

* Can be modified by the "administrator" of the "group", but cannot be modified by the ordinary "member",

* Can be read by the "members" of the "Group" (either in the same OOD or not) and synchronized to the corresponding location on the respective RootState

* Other members and "DEC" access the corresponding state through the "Member" RootState of "Group", with the same permissions as the RootState

@lurenpluto @waterflier

**Comment by lurenpluto** [comment]
At present, there is only one global-state of cyfs-stack, which is the state of the current protocol stack identity (People object usually), which is divided into root-state and local-cache
From the perspective of group requirements, a parallel group-global-state should be added, which belongs to the corresponding group identity; considering that cyfs-stack may support multiple identities in the future, that is, each identity will have a corresponding global-state

**Comment by lurenpluto** [comment]
Regarding the permissions control of group-state, group-state should only be a component and a set of internal interfaces provided by cyfs-stack, and will not directly provide RPC layer interfaces for external access; group components provide externally accessible RPC interfaces, And can directly access all group-states, so permission control should be implemented inside the group component for better

**Comment by streetycat** [comment]
Thank you, Is there someone can help me to support it. maybe it'll update the core code existed.

**Reply by lurenpluto** [reply]
> Thank you, Is there someone can help me to support it. maybe it'll update the core code existed.

https://github.com/buckyos/CYFS/issues/107#issuecomment-1429624952

Add a version of impls, u can see if it meets the requirements

**Comment by lurenpluto** [comment]
It has entered the development state, welcome to make suggestions and comments

https://github.com/buckyos/CYFS/issues/107

**Reply by streetycat** [reply]
Thank you, I have integrated It following the testcase

> It has entered the development state, welcome to make suggestions and comments
> 
> #107

Thank you, I have integrated It following the testcase.
